### ISSUE TITLE
Issues with the latest jupyter-client version 6.1.13

### ISSUE THREAD

**Issue by carlosperate** [mu-editor/mu#1444]
Released this mornig:
- https://github.com/jupyter/jupyter_client/releases/tag/6.1.13
- Diff: https://github.com/jupyter/jupyter_client/compare/6.1.12...6.1.13

These issues are reverted when downgrading to 6.1.12, but need to investigate further.

This is causing issues running the tests in Python 3.5:

```
$ python make.py check
============================= test session starts ==============================
platform darwin -- Python 3.5.10, pytest-6.1.2, py-1.10.0, pluggy-0.13.1 -- /Users/runner/hostedtoolcache/Python/3.5.10/x64/bin/python
cachedir: .pytest_cache
Using --random-order-bucket=module
Using --random-order-seed=978133

rootdir: /Users/runner/work/mu/mu, configfile: pytest.ini
plugins: random-order-1.0.4, cov-2.11.1, timeout-1.4.2
Unable to write settings to /Users/runner/Library/Application Support/mu/venv.json:
collecting ... collected 675 items / 13 errors / 662 selected
{

  "dirpath": "/Users/runner/Library/Application Support/mu/mu_venv-35-20210407-101553"
==================================== ERRORS ====================================
}
______________________ ERROR collecting tests/test_app.py ______________________
Traceback (most recent call last):
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
  File "/Users/runner/work/mu/mu/mu/settings.py", line 181, in save
    mod = import_path(self.fspath, mode=importmode)
    with open(saving_to_filepath, "w", encoding="utf-8") as f:
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
FileNotFoundError: [Errno 2] No such file or directory: '/Users/runner/Library/Application Support/mu/venv.json'
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/test_app.py:10: in <module>
    from mu.app import (
mu/app.py:49: in <module>
    from .modes import (
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
_______________ ERROR collecting tests/interface/test_dialogs.py _______________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/interface/test_dialogs.py:12: in <module>
    from mu.modes import (
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
________________ ERROR collecting tests/interface/test_main.py _________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/interface/test_main.py:11: in <module>
    from tests.test_app import DumSig
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/test_app.py:10: in <module>
    from mu.app import (
mu/app.py:49: in <module>
    from .modes import (
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
______________ ERROR collecting tests/modes/test_circuitpython.py ______________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_circuitpython.py:7: in <module>
    from mu.modes.circuitpython import CircuitPythonMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
__________________ ERROR collecting tests/modes/test_debug.py __________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_debug.py:6: in <module>
    from mu.modes.debugger import DebugMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
___________________ ERROR collecting tests/modes/test_esp.py ___________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_esp.py:5: in <module>
    from mu.modes.esp import ESPMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
__________________ ERROR collecting tests/modes/test_lego.py ___________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_lego.py:6: in <module>
    from mu.modes.lego import LegoMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
________________ ERROR collecting tests/modes/test_microbit.py _________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_microbit.py:10: in <module>
    from mu.modes.microbit import MicrobitMode, DeviceFlasher, can_minify
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
__________________ ERROR collecting tests/modes/test_pico.py ___________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_pico.py:6: in <module>
    from mu.modes.pico import PicoMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
_________________ ERROR collecting tests/modes/test_pyboard.py _________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_pyboard.py:7: in <module>
    from mu.modes.pyboard import PyboardMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
_______________ ERROR collecting tests/modes/test_pygamezero.py ________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_pygamezero.py:6: in <module>
    from mu.modes.pygamezero import PyGameZeroMode
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
_________________ ERROR collecting tests/modes/test_python3.py _________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_python3.py:7: in <module>
    from mu.modes.python3 import PythonMode, KernelRunner
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
___________________ ERROR collecting tests/modes/test_web.py ___________________
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/python.py:571: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/pathlib.py:517: in import_path
    importlib.import_module(module_name)
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:985: in _gcd_import
    ???
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:673: in _load_unlocked
    ???
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:171: in exec_module
    exec(co, module.__dict__)
tests/modes/test_web.py:7: in <module>
    from mu.modes.web import WebMode, CODE_TEMPLATE
mu/modes/__init__.py:1: in <module>
    from .python3 import PythonMode
mu/modes/python3.py:27: in <module>
    from qtconsole.manager import QtKernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/qtconsole/manager.py:9: in <module>
    from jupyter_client import KernelManager
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/__init__.py:6: in <module>
    from .asynchronous import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/__init__.py:1: in <module>
    from .client import AsyncKernelClient  # noqa
../../../hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/asynchronous/client.py:6: in <module>
    from jupyter_client.channels import HBChannel
E     File "/Users/runner/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
E       time_to_dead: float = 1.0
E                   ^
E   SyntaxError: invalid syntax
```

And also trying to run with Python 3.6:
```
$ python run.py 
No settings file found at /home/carlos/.local/share/mu/venv.json; skipping
Traceback (most recent call last):
  File "/home/carlos/workspace/mu/mu/modes/python3.py", line 372, in on_kernel_start
    self.view.add_jupyter_repl(kernel_manager, kernel_client)
  File "/home/carlos/workspace/mu/mu/interface/main.py", line 581, in add_jupyter_repl
    kernel_client.start_channels()
  File "/home/carlos/workspace/mu/venv/lib/python3.6/site-packages/qtconsole/kernel_mixins.py", line 47, in start_channels
    super(QtKernelClientMixin, self).start_channels(*args, **kw)
  File "/home/carlos/workspace/mu/venv/lib/python3.6/site-packages/jupyter_client/threaded.py", line 248, in start_channels
    super().start_channels(shell, iopub, stdin, hb, control)
  File "/home/carlos/workspace/mu/venv/lib/python3.6/site-packages/jupyter_client/client.py", line 295, in start_channels
    self.hb_channel.start()
  File "/home/carlos/workspace/mu/venv/lib/python3.6/site-packages/jupyter_client/client.py", line 364, in hb_channel
    self._hb_channel = self.hb_channel_class(self.context, self.session, url)
  File "/home/carlos/workspace/mu/venv/lib/python3.6/site-packages/qtconsole/util.py", line 65, in __new__
    QClass.__init__(inst)
TypeError: __init__() missing 1 required positional argument: 'context'
```

**Comment by carlosperate** [comment]
Related issues:
- https://github.com/jupyter/jupyter_client/issues/635
- https://github.com/jupyter/jupyter_client/issues/634
- https://github.com/jupyter/qtconsole/issues/476

**Comment by carlosperate** [comment]
Okay, so `jupyter-client` v6.1.13 has been yanked from PyPI and conda: https://pypi.org/project/jupyter-client/6.1.13/

So the CI should be passing again now, and the Python3 REPL in new Mu builds should be working again 🎉  

However:
- Next `jupyter-client` release will be v6.2 and won't support Python 3.5.
- We are currently using `qtconsole` v4, which does supports Python 3.5, but `qtconsole` v5 drops it
    - Small note to indicate that unfortunately the v5.0.0 release didn't set the `python_requires` version, and so pip in Py3.5 will (incorrectly) install it
- We should stay with `jupyter-client` < 6.2 and  `qtconsole` v4 for the Mu 1.1 final release
    - This would be mostly to support Debian (and specifically Raspbian) Stretch users and Ubuntu 16.04, both still supported, but not for long
    - After v1.1 is release we should finally drop Python 3.5
- I hope, but don't know yet, that `jupyter-client` v6.2 will fix the issues with `qtconsole` without `qtconsole` having to create a new release
    - If it does, then we wouldn't need to pin `jupyter-client`, as pip should detect the `python_requires` property and install the last compatible version
    - If it doesn't, it means there might be a `qtconsole` release, but that will be Python 3.6+
    - Either way is probably safer to lock `jupyter-client` to be < v6.2 and `qtconsole` to v4

Updated https://github.com/mu-editor/mu/pull/1445 with this in mind.

**Comment by carlosperate** [comment]
We are protected against these changes with #1443, which has been merged.
We'll likely remove it as soon as v1.1 final is released and drop Python 3.5
